apps/address-service/domains/common/utils/services/suggest/providers/DadataSuggestionProvider.js

```diff
@@ -199,21 +200,48 @@ class DadataSuggestionProvider extends AbstractSuggestionProvider {
      * @returns {Promise<*|null>}
      */
     async getOrganization (tin) {
-        const redis = getRedisClient('organization', 'tin')
+        return await this.callToDadataCached({
+            cacheClient: getRedisClient('organization', 'tin'),
+            searchKey: tin,
+            apiUrl: 'findById/party',
+            ttl: ORGANIZATION_TIN_CACHE_TTL,
+        })
+    }
+
+    /**
+     * @param {string} fiasId
+     * @returns {Promise<DadataObject|null>}
+     */
+    async getAddressByFiasId (fiasId) {
+        return await this.callToDadataCached({
+            cacheClient: getRedisClient('address', 'fiasId'),
+            searchKey: fiasId,
+            apiUrl: 'findById/address',
+            ttl: ADDRESS_TIN_CACHE_TTL,
+        })
+    }
 
-        const cached = await redis.get(tin)
+    /**
+     * @param {import('ioredis')} cacheClient
+     * @param {string} searchKey
+     * @param {string} apiUrl
+     * @param {number} ttl
+     * @returns {Promise<*|null>}
+     */
+    async callToDadataCached ({ cacheClient, searchKey, apiUrl, ttl = 3600 }) {
```

Comment:
Maybe move ttl value to local constant like `ORGANIZATION_TIN_CACHE_TTL `

Reply:
Done